src/binomial.ts

```diff
@@ -0,0 +1,129 @@
+import type {RngState} from "./state";
+
+/**
+ * Generate a binomially-distributed random number; the number of
+ * successes in `n` trials each with probability `p`. The
+ * implementation here switches between two different algorithms when
+ * `n * p` becomes large (10).
+ *
+ * @param state Random number state
+ *
+ * @param n Number of trials
+ *
+ * @param p Per-trial probability of success
+ */
+export function binomial(state: RngState, n: number, p: number) {
+    if (n === 0 || p === 0) {
+        return 0;
+    }
+    if (p === 1) {
+        return n;
+    }
+    const q = p <= 0.5 ? p : 1 - p;
+    const draw = n * q >= 10 ?
+        btrs(state, n, q) :
+        inversion(state, n, q);
+    return p <= 0.5 ? draw : n - draw;
+}
+
+function btrs(state: RngState, n: number, p: number) {
+    // This is spq in the paper.
```

Comment:
which paper? would be good to have a link to the source for these algorithms!

Reply:
hah, yeah, will copy that over :facepalm: